core/templates/domain/improvements/TaskEntryObjectFactory.ts

```diff
@@ -36,19 +36,19 @@ export interface ITaskEntryBackendDict {
   'resolved_on_msecs': number;
 }
 
-export interface ITaskEntryPayloadDict {
+export interface ITaskEntryPayloadDict<TaskType = string> {
   'entity_version': number;
-  'task_type': string;
+  'task_type': TaskType;
   'target_id': string;
   'issue_description': string;
   'status': string;
 }
 
-export class TaskEntry {
+export class TaskEntry<TaskType = string> {
```

Comment:
Shouldn't it be a abstract class? (Given that it's type depends on the parameter.) If yes, please add a comment as well here and above the interface.

Reply:
Nope, I'd like to instantiate these when the task type doesn't matter. For example, when displaying a list of these in the table, I'll render them using plain-old TaskEntry references instead.

Reply:
Ok, but I still think a comment should be there explained the `<TaskType = string>` especially.

Reply:
Done.